ichnaea/locate/provider.py

```diff
@@ -596,6 +600,17 @@ def should_locate(self, data, location):
     def locate(self, data):
         location = self.location_type(query_data=False)
 
+        if self.redis_client:
```

Comment:
if redis_client becomes a required argument, the condition here can check the ratelimit setting instead.